### PR TITLE
Correct delegate container implementation

### PR DESCRIPTION
**Pull request by elazar**
Refs container-interop/container-interop#76

### CODE REVIEW

**Comment by elazar** [comment]
@moufmouf @mnapoli If you wouldn't mind, I'd like your thoughts on these corrections to the delegate lookup feature implementation.

File: src/Container.php [hunk 1]
```diff
@@ -12,66 +11,40 @@
...
-     */
-    private function containerGet($id)
-    {
         try {
-            return $this->make($id);
-        } catch (InjectorException $previous) {
+            if ($this->delegate === null) {
+                return $this->make($id);
+            }
+            return $this->delegate->get($id);
```
Comment (mnapoli) [inline src/Container.php]: You shouldn't be fetching the entry from the delegate container. In `get()` you should fetch the entry from *your* container only. However dependencies of that entry should be resolved from the delegate.

**Comment by mnapoli** [comment]
Here is a very simple test that you can reuse for the delegate lookup feature : https://github.com/mnapoli/simplex/blob/master/src/Simplex/Tests/ContainerTest.php#L470-L480

**Comment by elazar** [comment]
@mnapoli Unfortunately, because the methods involved in dependency resolution within Auryn are private, they don't lend themselves to extension to support this sort of delegation. I'll remove it and amend container-interop/container-interop#76 accordingly.

**Comment by mnapoli** [comment]
I see, I had the same problem with Pimple and that's why I forked it and renamed it "simplex" :/

**Pull request merged**